### ISSUE TITLE
Rule to sort exports

### ISSUE THREAD

**Issue by dsaxton** [eslint/eslint#14155]
**Please describe what the rule should do:**

We have the `sort-imports` rule for sorting imports alphabetically, but there doesn't seem to be an analogous rule for _exports_. Have I simply overlooked this? If not I think it would be a useful addition.

**What new ECMAScript feature does this rule relate to?**

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[x] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
export {
  functionC,
  functionA,
  functionB,
}
```
should be reordered to
```js
export {
  functionA,
  functionB,
  functionC,
}
```

**Why should this rule be included in ESLint (instead of a plugin)?**

It's a natural counterpart to `sort-imports`.

**Are you willing to submit a pull request to implement this rule?**

I would if I knew how :smile:

**Comment by mdjermanovic** [comment]
Hi @dsaxton, thanks for the issue!

> We have the sort-imports rule for sorting imports alphabetically, but there doesn't seem to be an analogous rule for exports. Have I simply overlooked this? If not I think it would be a useful addition.

You're right, there's no core rule that sorts exports. Unfortunately, we had to make a decision to accept only new rules related to ECMAScript features added within the last 12 months, so this rule doesn't qualify. Please find more details in [this blog post](https://eslint.org/blog/2020/05/changes-to-rules-policies).

It's quite possible that this rule already exists in a plugin. Maybe you could check [eslint-plugin-sort-exports](https://www.npmjs.com/package/eslint-plugin-sort-exports), or file a new rule proposal on [eslint-plugin-import](https://github.com/benmosher/eslint-plugin-import).

I'm closing this issue per our rules policies, as we won't be able to accept this rule.
